cfme/tests/cloud/test_tenant.py

```diff
@@ -41,3 +61,159 @@ def test_tenant_crud(tenant):
+    """This function is used to check user permissions for particular tenant"""
+    view = navigate_to(appliance.collections.tenants, 'All')
+    count = view.table.row_count
+    for i in range(count):
+        existing_tenant = view.table.read()[i]['Name']
+        if assigned_tenant == existing_tenant:
+            row = view.table.row(name=existing_tenant)
+            row.click()
+            assert not view.toolbar.configuration.has_item('Manage Quotas')
+
+
+@test_requirements.quota
+@pytest.mark.ignore_stream('5.9')
+@pytest.mark.tier(1)
+def test_dynamic_product_feature_for_tenant_quota(request, appliance, new_tenant):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/12h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: Configuration
+        tags: quota
+        testSteps:
+            1. Add two users > alpha and omega
+            2. Create two tenants > alpha_tenant and it's child - omega_tenant
+            3. Create two custom roles (role_alpha and role_omega) from copying
+               EvmRole-tenant-administrator role
+            4. Create groups alpha_group(for alpha_tenant) and omega_group(for omega_tenant)
+               then assign role_alpha to alpha_group and role_omega to omega_group
+            5. Add alpha_group to alpha user and omega_group to omega user
+            6. Modify role_alpha for manage quota permissions of alpha user as it will manage
+               only quota of omega_tenant
+            7. Modify role_omega for manage quota permissions of omega user as it will not even
+               manage quota of itself or other tenants
+            8. CHECK IF YOU ARE ABLE TO MODIFY THE "MANAGE QUOTA" CHECKS IN ROLE AS YOU WANT
+            9  Then see if you are able to save these two new roles.
+            10. Then login with both users one by one and see if these roles permissions are
+                applied or not.
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5.
+            6.
+            7.
+            8.
+            9.
+            10. Logged in with users created
+            10a. login with alpha and SEE IF ALPHA USER CAN ABLE TO SET QUOTA OF omega_tenant
+                 FOR CPU - 5('Manage Quotas' option should be available)
+            11b. login with omega and SEE QUOTA GETS CHANGED OR NOT. THEN TRY TO CHANGE QUOTA
+                 IMPOSED BY ALPHA USER.
+            12c. Here as per role_omega permissions, omega must not able change its own quota or
+                 other tenants quota.
+
+    Bugzilla:
+        1655012, 1468795
+    """
+    # Here TENANT[1] is the second tenant(child tenant) from the tenant's list
+    child_tenant = appliance.collections.tenants.create(
+        name="{}".format(TENANTS[1]),
+        description="tenant_des{}".format(fauxfactory.gen_alphanumeric()),
```

Comment:
I would prefer using the `child_tenant` name somewhere in the description instead of some randomly generated data. Totally optional though.